pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -140,9 +89,9 @@ class XarrayZarrRecipe(BaseRecipe):
     file_pattern: FilePattern
     inputs_per_chunk: Optional[int] = 1
     target_chunks: Dict[str, int] = field(default_factory=dict)
-    target: AbstractTarget = field(default_factory=UninitializedTarget)
-    input_cache: AbstractTarget = field(default_factory=UninitializedTarget)
-    metadata_cache: AbstractTarget = field(default_factory=UninitializedTarget)
+    target: Optional[AbstractTarget] = None
```

Comment:
Is the decision to define a default with `field(default_factory=...)` vs. as `None` a consequence of whether or not we want the user to be able to access certain methods of a default type vs. just overwrite the default value?

Reply:
No. TBH I can't remember the original reason why I thought I had to use `UninitializedTarget`. It may have had to do with type hints. Before, all these objects _had_ to be `AbstractTarget`. Now they are `Optional[AbstractTarget]` or `Optional[MetadataTarget]`, etc. The `Optional` type hint means that `None` is a valid value. This is simpler and more in line with standard python best practices.

One downstream effect, however, is that we need statements like `if self.target is not None` in various places in order to keep mypy happy.